logstreams/src/main/java/io/zeebe/logstreams/processor/StreamProcessorController.java

```diff
@@ -294,9 +294,16 @@ private void processEvent(final LoggedEvent event) {
       eventProcessor = streamProcessor.onEvent(event);
 
       if (eventProcessor != null) {
-        zeebeDb.transaction(eventProcessor::processEvent);
-        metrics.incrementEventsProcessedCount();
-        actor.runUntilDone(this::executeSideEffects);
+        try {
+          zeebeDb.transaction(eventProcessor::processEvent);
+          metrics.incrementEventsProcessedCount();
+          actor.runUntilDone(this::executeSideEffects);
+        } catch (final Exception e) {
+          LOG.error(ERROR_MESSAGE_PROCESSING_FAILED_SKIP_EVENT, getName(), event, e);
+          zeebeDb.transaction(() -> eventProcessor.processingFailed(e));
+          // send rejection etc
+          actor.runUntilDone(this::executeSideEffects);
```

Comment:
Is there a risk that we also execute side effects such as sending a `CORRELATE` message command, or other undesirable things?

Reply:
It will not since it is rested in the processingFailed